src/string.cr

```diff
@@ -3312,15 +3312,20 @@ class String
   # Update rolling hash for Rabin-Karp algorithm `String#index`.
   private macro update_hash(n)
     {% for i in 1..n %}
-      {% if i != 1 %}
-        byte = head_pointer.value
-      {% end %}
+      byte = head_pointer.value
```

Comment:
Because of this, I think the loop body inside `#byte_index(String)` (around line 3784) can now be expressed as `update_hash 1`

Reply:
Seems reasonable.
I'll rebase branch on this week, and will accept your suggestion.

Reply:
fixed